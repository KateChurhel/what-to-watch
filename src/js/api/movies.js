/* istanbul ignore file */
import { sendGetRequest } from './index';

export const getCategoryData = (category, data) => sendGetRequest(`/${category}/popular`, data);

export const getDetailsData = (category, id, data) => sendGetRequest(`/${category}/${id}`, data);

export const getSearchResult = (category, data) => sendGetRequest(`/search/${category}`, data);

export const getDiscoverResult = (data) => sendGetRequest('/discover/movie', data);

export const getGenres = () => sendGetRequest('/genre/movie/list');
