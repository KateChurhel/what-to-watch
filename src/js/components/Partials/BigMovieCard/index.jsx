// libraries
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
// constants
import { URL_IMAGE } from '../../../constants/url';
import CATEGORIES from '../../../constants/categories';

const BigMovieCard = ({
  id, poster, backgroundImage, formattedRunTime, releaseYear, genresList, cardTitle, tagline,
}) => (id ? (
  <section className="movie-card">
    <div
      className="movie-card__bg"
      style={{ backgroundImage: `url('${URL_IMAGE.backgroundImage}${backgroundImage}')` }}
    />

    <div className="container">
      <div className=" movie-card__wrap">
        <div className="movie-card__info movie-card__info--main">
          <div className="movie-card__poster movie-card__poster--main">
            <img alt={cardTitle} loading="lazy" src={`${URL_IMAGE.poster}${poster}`} />
          </div>

          <div className="movie-card__desc">
            <h2 className="movie-card__title">{`${cardTitle} (${releaseYear})`}</h2>
            <p className="movie-card__meta">
              <span className="movie-card__genre">{genresList}</span>
              <span className="movie-card__runtime">{formattedRunTime}</span>
            </p>
            <i className="movie-card__tagline movie-card__tagline--main">{tagline}</i>

            <Link className="button movie-card__button" to={`/details/${CATEGORIES.movie}/${id}`}>More Info</Link>
          </div>
        </div>
      </div>
    </div>
  </section>
) : null);

BigMovieCard.defaultProps = {
  id: 0,
  poster: '',
  backgroundImage: '',
  formattedRunTime: '',
  releaseYear: '',
  genresList: '',
  cardTitle: '',
  tagline: '',
};

BigMovieCard.propTypes = {
  id: PropTypes.number,
  poster: PropTypes.string,
  backgroundImage: PropTypes.string,
  formattedRunTime: PropTypes.string,
  releaseYear: PropTypes.string,
  genresList: PropTypes.string,
  cardTitle: PropTypes.string,
  tagline: PropTypes.string,
};

export default BigMovieCard;
