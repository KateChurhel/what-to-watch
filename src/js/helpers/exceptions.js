// constants
import DEFAULT_ERROR from '../constants/errors';

export class RequestError extends Error {
  constructor(status = '', responseData) {
    super();

    const errorData = responseData || {};

    this.message = errorData.message || DEFAULT_ERROR;
    this.errors = errorData.errors;
    this.status = status;
    this.name = 'RequestError';
  }
}

export default {};
