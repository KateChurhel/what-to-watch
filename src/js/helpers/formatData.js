import moment from 'moment';
import 'moment-duration-format';

export const formatTime = (time, unit, format) => moment.duration(time, unit).format(format, { trim: false });
export const formatDate = (date, format) => moment(date).format(format);

export const parseQueryString = (queryString) => {
  const query = {};
  const pairs = (queryString[0] === '?' ? queryString.substr(1) : queryString).split('&');
  for (let i = 0; i < pairs.length; i++) {
    const pair = pairs[i].split('=');
    query[decodeURIComponent(pair[0])] = decodeURIComponent(pair[1] || '');
  }

  return query;
};

export const getStringValues = (array = [], key) => (array.length ? array.map((item) => item[key]).join(', ') : '');

export const formatCardData = ({
  number_of_episodes: numberOfEpisodes, number_of_seasons: numberOfSeasons, original_name: originalName,
  episode_run_time: episodeRunTime, reviews: { results: reviews = [] } = {}, homepage, original_title: originalTitle,
  original_language: originalLanguage, spoken_languages: spokenLanguages, status, runtime, tagline,
  release_date: releaseDate, genres, poster_path: poster, backdrop_path: backgroundImage,
  vote_average: voteAverage, vote_count: voteCount, overview, title, name, id,
  credits: { cast = [] } = {}, recommendations: { results: recommendations = [] } = {},
} = {}) => {
  const formattedRunTime = formatTime(runtime || episodeRunTime, 'minutes', 'h[h] m[m]');
  const releaseYear = formatDate(releaseDate, 'YYYY');
  const genresList = getStringValues(genres, 'name');
  const spokenLanguagesList = getStringValues(spokenLanguages, 'name');

  return {
    id,
    numberOfEpisodes,
    numberOfSeasons,
    episodeRunTime,
    reviews,
    homepage,
    originalLanguage,
    status,
    tagline,
    releaseDate,
    poster,
    backgroundImage,
    voteAverage,
    voteCount,
    overview,
    cast,
    recommendations,
    originalTitle: originalTitle || originalName,
    cardTitle: title || name,
    formattedRunTime,
    releaseYear,
    genresList,
    spokenLanguagesList,
  };
};
