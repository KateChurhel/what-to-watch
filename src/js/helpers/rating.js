import USER_RATINGS from '../constants/rating';

export const computeRatingLevel = (rating) => USER_RATINGS.filter(
  (rank) => rank.min <= rating,
).pop().title;
