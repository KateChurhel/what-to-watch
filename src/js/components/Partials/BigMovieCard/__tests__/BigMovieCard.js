import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import BigMovieCard from '../index';
import { formatCardData } from '../../../../helpers/formatData';
import { MOCK_MOVIES } from '../../../../mocks/movies';

describe('Big Movie Card: ', () => {
  it('correctly renders with empty data', () => {
    const tree = shallow(<BigMovieCard />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('correctly renders with card data', () => {
    const tree = shallow(<BigMovieCard {...formatCardData(MOCK_MOVIES[0])} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
