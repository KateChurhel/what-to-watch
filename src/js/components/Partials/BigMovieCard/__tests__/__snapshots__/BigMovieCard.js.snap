// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Big Movie Card:  correctly renders with card data 1`] = `
<section
  className="movie-card"
>
  <div
    className="movie-card__bg"
    style={
      Object {
        "backgroundImage": "url('https://image.tmdb.org/t/p/original/qJeU7KM4nT2C1WpOrwPcSDGFUWE.jpg')",
      }
    }
  />
  <div
    className="container"
  >
    <div
      className=" movie-card__wrap"
    >
      <div
        className="movie-card__info movie-card__info--main"
      >
        <div
          className="movie-card__poster movie-card__poster--main"
        >
          <img
            alt="La La Land"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w300/uDO8zWDhfWwoFdKS4fzkUJt0Rf0.jpg"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            La La Land (2016)
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Comedy, Drama, Romance, Music
            </span>
            <span
              className="movie-card__runtime"
            >
              2h 9m
            </span>
          </p>
          <i
            className="movie-card__tagline movie-card__tagline--main"
          >
            Here's to the fools who dream.
          </i>
          <a
            className="button movie-card__button"
            href="/details/movie/313369"
          >
            More Info
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Big Movie Card:  correctly renders with empty data 1`] = `""`;
